### PR TITLE
[BottomNavigation] Add color themer to bottom navigation bar

### PR DESCRIPTION
**Pull request by jgunaratne**
Add color themer to bottom navigation bar.

Pending merge of: https://github.com/material-components/material-components-ios/pull/2088

simulator screen shot - iphone 8 - 2017-10-20 at 17 14 51

### CODE REVIEW

**Comment by ianegordon** [comment]
Blocked on #2088
Will revisit as soon as BottomNavigationBar lands.

**Comment by ianegordon** [comment]
@romoore could you take a look at this when BottomNavigationBar lands?

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (romoore) [inline components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h]: I think it would be good to provide a description of what the themer will do (and what it won't do). For example, does it apply `tintColor`, `backgroundColor`, selected item color, etc.?  Are there any colors that aren't changed/updated that clients should expect to change manually?

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (romoore) [inline components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h:33]: Does this need to be `NSObject` or can it just be `id`?

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.m
Comment (romoore) [inline components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.m]: Nit: Same indentation concern.

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (romoore) [inline components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h]: Nit: I think this needs to be indented 1 more space. It's at 3, but I believe it should be 4.

**Review by romoore — CHANGES_REQUESTED** [review]
If we have a Swift example, would love to see it themed as well. If not, we can add it later.

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.m
Comment (jgunaratne) [reply components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.m]: Done.

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (jgunaratne) [reply components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h]: Done.

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (jgunaratne) [reply components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h:33]: All of the other themers use NSObject<MDCColorScheme>. Would be good to have them all be consistent.

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (jgunaratne) [reply components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h]: I've elaborated on how this affects the various color properties of the bottom navigation bar. The client shouldn't have to change any colors manually when using the themer.

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (romoore) [reply components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h:33]: Ack. Sounds like a good reason. :)

**Review by romoore — APPROVED** [review]

File: components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h
Comment (jgunaratne) [reply components/BottomNavigation/src/ColorThemer/MDCBottomNavigationBarColorThemer.h:33]: Also added theming to the Swift example as suggested.

**Pull request merged**
